dlib/image_transforms/interpolation.h

```diff
@@ -229,21 +229,22 @@ namespace dlib
     class interpolate_bilinear
     {
     public:
-
         template <typename T, typename image_view_type, typename pixel_type>
-        typename disable_if<is_rgb_image<image_view_type>,bool>::type operator() (
+        bool operator() (
             const image_view_type& img,
-            const dlib::vector<T,2>& p,
+            const dlib::vector<T, 2>& p,
             pixel_type& result
         ) const
         {
-            COMPILE_TIME_ASSERT(pixel_traits<typename image_view_type::pixel_type>::has_alpha == false);
+            // Interpolation currently supports only fully cartesian (non-polar) spaces.
+            COMPILE_TIME_ASSERT(is_color_space_cartesian_image<image_view_type>::value == true);
 
-            const long left   = static_cast<long>(std::floor(p.x()));
-            const long top    = static_cast<long>(std::floor(p.y()));
-            const long right  = left+1;
-            const long bottom = top+1;
+            using traits = pixel_traits<typename image_view_type::pixel_type>;
 
+            const long left = static_cast<long>(std::floor(p.x()));
+            const long top = static_cast<long>(std::floor(p.y()));
+            const long right = left + 1;
+            const long bottom = top + 1;
```

Comment:
Please don't make white space only changes.  It makes the pull request hard to reivew.